camus-sweeper/src/main/java/com/linkedin/camus/sweeper/CamusHourlySweeper.java

```diff
@@ -0,0 +1,327 @@
+
+    return executorService.submit(new KafkaCollectorRunner(jobName, props, errorMessages, topic));
+  }
+
+  @Override
+  protected void runCollectorForTopicDir(FileSystem fs, String topic, Path topicSourceDir, Path topicDestDir)
+      throws Exception {
+    LOG.info("Running collector for topic " + topic + " source:" + topicSourceDir + " dest:" + topicDestDir);
+    List<Future<?>> tasksToComplete = new ArrayList<Future<?>>();
+
+    List<Properties> jobPropsList = planner.createSweeperJobProps(topic, topicSourceDir, topicDestDir, fs, metrics);
+
+    for (Properties jobProps : jobPropsList) {
+      tasksToComplete.add(runCollector(jobProps, topic));
+    }
+  }
+
+  private class KafkaCollectorRunner extends CamusSweeper.KafkaCollectorRunner {
+
+    public KafkaCollectorRunner(String name, Properties props, List<SweeperError> errorQueue, String topic) {
+      super(name, props, errorQueue, topic);
+    }
+
+    @Override
+    public void run() {
+      KafkaCollector collector = null;
+      try {
+        LOG.info("Starting runner for " + this.props.getProperty(TOPIC_AND_HOUR));
+        collector = new KafkaCollector(props, name, topic);
+        LOG.info("Running " + name + " for input " + props.getProperty(INPUT_PATHS));
+        collector.run();
+      } catch (Throwable e) { // Sometimes the error is the Throwable, e.g. java.lang.NoClassDefFoundError
+        e.printStackTrace();
+        LOG.error("Failed for " + name + " ,job: " + collector == null ? null : collector.getJob() + " failed for "
+            + props.getProperty(INPUT_PATHS) + " Exception:" + e.getLocalizedMessage());
+        errorQueue.add(new SweeperError(name, props.get(INPUT_PATHS).toString(), e));
+      }
+    }
+  }
+
+  private class KafkaCollector extends CamusSweeper.KafkaCollector {
+
+    private final String topicAndHour;
+    private final Path[] inputPaths;
+    private final Path tmpPath;
+    private final Path outputPath;
+    private final FileSystem fs;
+
+    public KafkaCollector(Properties props, String jobName, String topicName) throws IOException {
+      super(props, jobName, topicName);
+      this.topicAndHour = props.getProperty(TOPIC_AND_HOUR);
+      this.fs = FileSystem.get(job.getConfiguration());
+      this.inputPaths = getInputPaths();
+      this.tmpPath = new Path(job.getConfiguration().get(TMP_PATH));
+      this.outputPath = new Path(job.getConfiguration().get(DEST_PATH));
+      addInputAndOutputPathsToFileInputFormat();
+    }
+
+    private Path[] getInputPaths() {
+      List<String> strPaths = Utils.getStringList(props, INPUT_PATHS);
+      Path[] inputPaths = new Path[strPaths.size()];
+      for (int i = 0; i < strPaths.size(); i++)
+        inputPaths[i] = new Path(strPaths.get(i));
+      return inputPaths;
+    }
+
+    @Override
+    public void run() throws Exception {
+      CamusHourlySweeper.this.metrics.runnerStartTimeByTopic.put(this.topicAndHour, System.currentTimeMillis());
```

Comment:
Rather than directly accessing these instance variables from the metrics class, it would be clearer if you have a method such as: reportHourlyRunnerStartTime(String topic, String hour, long startTime)